tests/integration/test_remoted/test_agent_communication/test_agent_version_shared_configuration_startup_message.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+import os
+from time import sleep
+
+import pytest
+import wazuh_testing.tools.agent_simulator as ag
+from wazuh_testing import UDP, TCP, TCP_UDP, remote
+from wazuh_testing.remote import check_push_shared_config
+from wazuh_testing.tools import LOG_FILE_PATH
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.monitoring import FileMonitor
+
+pytestmark = pytest.mark.tier(level=2)
+
+# Configuration
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_agent_version_shared_configuration_startup_message.yaml')
+agent_conf_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data', 'agent.conf')
+
+parameters = [
+    {'PROTOCOL': TCP},
+    {'PROTOCOL': UDP},
+    {'PROTOCOL': TCP_UDP}
+]
+
+metadata = [
+    {'protocol': TCP},
+    {'protocol': UDP},
+    {'protocol': TCP_UDP}
+]
+
+agent_info = {
+    'debian7_420': {
+        'manager_address': '127.0.0.1',
+        'os': 'debian7',
+        'version': 'v4.2.0',
+        'disable_all_modules': True
+    },
+    'debian9_4.4.0': {
+        'manager_address': '127.0.0.1',
+        'os': 'debian9',
+        'version': 'v4.4.0',
+        'disable_all_modules': True
+    }
+}
+
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+config_ids = [x['PROTOCOL'] for x in parameters]
+
+
+# fixtures
+@pytest.fixture(scope="module", params=configurations, ids=config_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.mark.parametrize("agent_name", agent_info.keys())
+def test_agent_remote_configuration(agent_name, get_configuration, configure_environment, remove_shared_files,
+                                    restart_remoted, create_agent_group):
+    """Check if the agents correctly send their version, receive the shared configuration, and finally,
+        the start-up message is received and processed by the manager.
+
+    Raises:
+        AssertionError: if `wazuh-db` returns a wrong agent version, agents do not receive shared configuration or
+        startup message after agent restart is not received.
+    """
+
+    protocols = get_configuration['metadata']['protocol']
+
+    for protocol in protocols.split(","):
+        agent = ag.Agent(**agent_info[agent_name])
+        sender = ag.Sender(agent_info[agent_name]['manager_address'], protocol=protocol)
+
+        # Sleep to avoid ConnectionRefusedError
+        sleep(1)
```

Comment:
```suggestion
        agent = ag.Agent(**agent_info[agent_name])

        # Sleep to avoid ConnectionRefusedError
        sleep(1)
        
        sender = ag.Sender(agent_info[agent_name]['manager_address'], protocol=protocol)
```

Reply:
I've tested this and it is true that sometimes defining the sender before the sleep could cause errors so we've moved it as suggested by @jmv74211. The change was introduced within an amend and a force-push.